`Bug` [Samsung_HVAC]: deprecated function calls

### The problem

```
src/esphome/components/samsung_ac/samsung_ac_device.cpp: In member function 'virtual esphome::climate::ClimateTraits esphome::samsung_ac::Samsung_AC_Climate::traits()':
src/esphome/components/samsung_ac/samsung_ac_device.cpp:37:46: warning: 'void esphome::climate::ClimateTraits::set_supported_custom_fan_modes(const char* const (&)[N]) [with unsigned int N = 1]' is deprecated: Call set_supported_custom_fan_modes() on the Climate entity instead. Removed in 2026.11.0 [-Wdeprecated-declarations]
   37 |         traits.set_supported_custom_fan_modes(CUSTOM_FAN_MODES);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
In file included from src/esphome/components/climate/climate.h:11,
                 from src/esphome/components/samsung_ac/samsung_ac_device.h:12,
                 from src/esphome/components/samsung_ac/samsung_ac.h:9,
                 from src/esphome/components/samsung_ac/samsung_ac_device.cpp:2:
src/esphome/components/climate/climate_traits.h:118:8: note: declared here
  118 |   void set_supported_custom_fan_modes(const char *const (&modes)[N]) {
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/samsung_ac/samsung_ac_device.cpp:61:48: warning: 'void esphome::climate::ClimateTraits::set_supported_custom_presets(const std::vector<const char*, std::allocator<const char*> >&)' is deprecated: Call set_supported_custom_presets() on the Climate entity instead. Removed in 2026.11.0 [-Wdeprecated-declarations]
   61 |             traits.set_supported_custom_presets(custom_presets);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
src/esphome/components/climate/climate_traits.h:150:8: note: declared here
  150 |   void set_supported_custom_presets(const std::vector<const char *> &presets) {
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### ESPHome Version

2026.6.0

### Samsung HVAC Component Version

github://omerfaruk-aran/esphome_samsung_hvac_bus@main

### Home Assistant Version

2026.6.3

### AC Type

OTHER

### Indoor AC Unit Model

*

### Outdoor AC Unit Model

*

### ESP Device Model

*

### Connection Points

*

### ESPHome Logs

_No response_

### Samsung AC Logs

_No response_

### YAML Configuration

```yaml

```

### Additional Details

_No response_

--- a/components/samsung_ac/samsung_ac_device.cpp
+++ b/components/samsung_ac/samsung_ac_device.cpp
@@ -34,7 +34,7 @@ namespace esphome
                                         climate::CLIMATE_FAN_AUTO});
 
         static const char *CUSTOM_FAN_MODES[] = {"Turbo"};
-        traits.set_supported_custom_fan_modes(CUSTOM_FAN_MODES);
+        this->set_supported_custom_fan_modes(CUSTOM_FAN_MODES);
       }
 
       {
@@ -58,7 +58,7 @@ namespace esphome
 
           if (!custom_presets.empty())
           {
-            traits.set_supported_custom_presets(custom_presets);
+            this->set_supported_custom_presets(custom_presets);
           }
         }
       }
@@ -200,7 +200,3 @@ namespace esphome
 
   } // namespace samsung_ac
 } // namespace esphome
-
-
-
-

